### PR TITLE
[xc-admin] enable publisher key and multisig signer key mapping when available

### PR DESCRIPTION
**Pull request by cctdaniel**
according to @thmzlt the mapping will be accessible within the container as a json file and hence the approach (for local dev we can create our own json file with the same name but add it to .gitignore so that we dont commit them to the repo)

### CODE REVIEW

File: governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals.tsx [hunk 1]
```diff
@@ -425,10 +429,20 @@ const Proposal = ({
...
-            <div className="flex justify-between" key={pubkey.toBase58()}>
-              <div>Key {idx + 1}</div>
-              <CopyPubkey pubkey={pubkey.toBase58()} />
-            </div>
+            <>
+              <div className="flex justify-between" key={pubkey.toBase58()}>
+                <div>Key {idx + 1}</div>
+                <CopyPubkey pubkey={pubkey.toBase58()} />
+              </div>
+              {pubkey.toBase58() in multisigSinerKeyToNameMapping ? (
```
Comment (guibescos) [inline governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals.tsx:437]: multisigSinerKeyToNameMapping -> multisigSignerKeyToNameMapping

File: governance/xc_admin/packages/xc_admin_frontend/pages/index.tsx [hunk 2]
```diff
@@ -10,6 +11,22 @@ import { MultisigContextProvider } from '../contexts/MultisigContext'
 import { PythContextProvider } from '../contexts/PythContext'
 import { classNames } from '../utils/classNames'
 
+export const getStaticProps: GetStaticProps = async () => {
+  const publisherKeyToNameMapping = fs.existsSync('publishers.json')
+    ? JSON.parse((await fs.promises.readFile('publishers.json')).toString())
```
Comment (guibescos) [inline governance/xc_admin/packages/xc_admin_frontend/pages/index.tsx:16]: I couldn't help but notice that this mapping is the opposite of reference_data : (ref_data is `name -> key` and this is `key-> name` )

**Review by guibescos — APPROVED** [review]

File: governance/xc_admin/packages/xc_admin_frontend/pages/index.tsx (hunk 2 above)
Comment (thmzlt) [reply governance/xc_admin/packages/xc_admin_frontend/pages/index.tsx:16]: The map that will be fed is key -> name

File: governance/xc_admin/packages/xc_admin_frontend/next.config.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ const nextConfig = {
     externalDir: true,
   },
   webpack(config) {
+    config.resolve.fallback = { fs: false }
```
Comment (thmzlt) [inline governance/xc_admin/packages/xc_admin_frontend/next.config.js:8]: How does this work?

**Comment by thmzlt** [comment]
I understand you guys are collaborating on this separately, but it would be nice to have instructions on the README on how to run this. I couldn't get it up to test passing the publishers/signers files.

File: governance/xc_admin/packages/xc_admin_frontend/next.config.js (hunk 3 above)
Comment (cctdaniel) [reply governance/xc_admin/packages/xc_admin_frontend/next.config.js:8]: this is required for calling fs in next.js

File: governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals.tsx (hunk 1 above)
Comment (cctdaniel) [reply governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals.tsx:437]: lmao Siner

**Pull request merged**